app/views/partials/channels-tab-access-control.html

```diff
@@ -1,7 +1,28 @@
-<div class="form-group" ng-class="{ 'has-error' : ngError.allow }">
-  <label><span>*</span> Allowed roles and clients</label>
-  <taglist tag-data="channel.allow" ><input class="form-control" placeholder="Enter a role and press enter..." type="text" ng-model="allowedRoles" typeahead="role for role in taglistClientRoleOptions | filter:$viewValue" /></taglist>
-  <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>{{validationRequiredMsg}}</div>
+<div>
+  <span>
+    <label> Directly Assigned Clients</label><br/>
+    <ul>
+      <li ng-repeat="client in assignedClients"><span >{{client}}</span></li>
+    </ul>
+  </span>
+</div>
+
+<div class="form-group-2columns" ng-class="{ 'has-error' : ngError.roles }">
+  <div class="form-group" style="float: left;" ng-class="{ 'has-error' : ngError.password }">
+    <label><span>**</span> Assign Existing Roles</label>
+    <br/>
+    <span ng-repeat="role in roles">
+      <input type="checkbox" ng-model="requiredForNgChange" ng-checked="formData.assigned[role.name]" ng-change="toggleAssignedRoles(role.name)">{{role.name}}</input>
+      <br/>
+    </span>
+    <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>Either '<strong>Assign Existing Roles</strong>' or '<strong>Add New Role</strong>' is required!</div>
+  </div>
+  <div class="form-group" style="float: right;" ng-class="{ 'has-error' : ngError.roles }">
+    <label><span>**</span> Add New Role</label>
+    <input class="form-control" type="text" name="newRole" ng-model="formData.newChannelRole" ng-blur="createNewRole()" ng-keyup="$event.keyCode == 13 && createNewRole()"/>
+    <div style="color:orange" ng-show="formData.duplicateNewRole"><i class="glyphicon glyphicon-warning-sign"></i>Role/Client already assigned to Channel</div>
+    <div class="has-error-msg"><i class="glyphicon glyphicon-warning-sign"></i>Either '<strong>Assign Existing Roles</strong>' or '<strong>Add New Role</strong>' is required!</div>
+  </div>
```

Comment:
We want to move away from adding clients directly to channels.  You can still add a client by typing its name in the add new role input.